canonical/cloud-init
# first commit for first cloud-init PR

## Proposed Commit Message


```
Script handlers for per-boot, per-instance, and per-once.

Several (6?) months ago I inquired on #cloud-init about how to include multiple init scripts in user data, one per instance and one per boot. I was told this wasn't supported. I needed the behavior, so I had a discussion about implementation, and when I suggested a simple custom part handler for each frequency, to be used during the normal make-mime process, I was told that sounded good and it might be supported.

I've been using my solution in my own work for months, and since a client is about to go live with it, it seems like a good time to work towards getting it included in cloud-init proper.
```

## Additional Context


## Test Steps


I'm currently running a command like the following

```
cloud-init devel make-mime --force \
                                --attach "$myCloudInit/cloudinit/cmd/devel/shell_script_per_boot.py":part-handler \
                                --attach "$myCloudInit/cloudinit/cmd/devel/shell_script_per_instance.py":part-handler \
                                --attach "$myCloudInit/cloudinit/cmd/devel/shell_script_per_once.py":part-handler \
                                --attach "$myScriptFolder./init-boot.sh":x-shellscript-per-boot \
                                --attach" $myScriptFolder/init-instance.sh":x-shellscript-per-instance
```                                
Then, I deploy the file as used data, check logfiles, and check /var/lib/cloud/scripts/per-boot and -instance to make sure my files are there.

I'd love for the script files to be placed in cloud-init such that they do not need to be explicitly added to userdata. But I didn't know how to do that, so I put them in cloudinit/cmd/devel instead.

## Checklist:


(I am happy to do these once my code looks like it's on the right track)
